component/org.wso2.carbon.identity.oauth2.token.handler.clientauth.mutualtls/src/main/java/org/wso2/carbon/identity/oauth2/token/handler/clientauth/mutualtls/MutualTLSClientAuthenticator.java

```diff
@@ -177,7 +180,17 @@ public boolean canAuthenticate(HttpServletRequest request, Map<String, List> bod
                                    OAuthClientAuthnContext context) {
 
         String headerName = IdentityUtil.getProperty(CommonConstants.MTLS_AUTH_HEADER);
-        if (clientIdExistsAsParam(bodyParams)) {
+        if (clientIdExistsAsParam(bodyParams) || clientIdExistsAsParameter(request)) {
+            try {
+                if (!isMTLSAuthentication(request, bodyParams)) {
```

Comment:
Change the authenticators hitting orders and add a flag to check the valid authenticator